cpp/splinepy/utils/arrays.hpp

```diff
@@ -210,13 +210,14 @@ GaussWithPivot(std::array<std::array<double, para_dim>, para_dim>& A,
 
   // back substitution
   double sum;
-  for (int i{para_dim - 1}; i >= 0; i--) {
+  for (std::size_t i_para_dim{}; i_para_dim < para_dim; i_para_dim++) {
+    std::size_t i{para_dim - 1 - i_para_dim};
```

Comment:
what kind of warning was this causing?

Reply:
This was causing signed unsigned comparison. In order to make `i_para_dim` an unsigned type, I had to change the order, as unsigned types are always `>=0` making the loop unbreakable. So I just introduced a new variable for indexing

Reply:
is there warning if you keep the `int`?

Reply:
Yes in a later loop. However, I have a feeling that you are not happy with this implementation. There are two alternatives:
```
for (std::size_t i{para_dim - 1}; i != static_cast<std::size_t>(-1) ; i--)
or
for (std::size_t i{para_dim - 1}; i-- > 0;)
```
Maybe I can convince you of either of these

Reply:
I guess you just know how to read my mind. I am already convinced. The second one: did you already know that too?

Reply:
Haha :D I kind of regret suggesting the second one, as I find it harder to read because of the first iteration and also it looks really bad. Any chance we can settle on the first one as a compromise?

Reply:
hey hey, no backsies! :D  
For me, the second one is clear. However the first one, `static_cast<std::size_t>(-1)` confuses me a bit

Reply:
What you mean to say is "you annoyed me so much that I gave in", but i still take it 😆